src/main/java/externalServices/video_processing_service/VideoProcessingService.java

```diff
@@ -0,0 +1,83 @@
+package externalServices.video_processing_service;
+import com.github.kokorin.jaffree.ffmpeg.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+public class VideoProcessingService implements IVideoProcessingService{
+    @Override
+    public void mp4ToHls(String mp4) {
+        Path input = Paths.get("src/main/resources/raw/" + mp4);
+        String baseName = mp4.substring(0, mp4.lastIndexOf('.'));
+        Path outputDir = Paths.get("src/main/resources/encoded/" + baseName);
+
+        Path masterPlaylist = outputDir.resolve("master.m3u8");
+        if (Files.exists(masterPlaylist)) {
+            System.out.println("Video has already been encoded.");
+            return;
+        }
```

Comment:
current handling if the file exists is to not encode, but this doesn't allow people to overwrite, this is design choice, might want to consider to just overwrite the whole, can be considered and keep in mind